Gem and me0 segment bug fix 81x

jshlee opened the pull request:
ME0Segments changed reference detector from etaPartition to chamber
* ME0SegmentAlgorithm.cc - changed ME0EtaPartition -> ME0Chamber, bug fix for failed fits
* ME0SegmentAlgorithmBase.h - changed ME0EtaPartition -> ME0Chamber
* ME0SegmentBuilder.cc  - changed ME0EtaPartition -> ME0Chamber
* ME0SegmentMatcher.cc - changed ME0EtaPartition -> ME0Chamber
* ME0SegmentsValidation.cc - changed ME0EtaPartition -> ME0Chamber

muonME0DigisPreReco_cfi.py - etaResolution was set to 0, this gave me0RecHits 0 for y error, which made ME0Segment fitting fail.

MuonSegFit - increased number of hits that can be fitted

MuonGEMDetLayerGeometryBuilder.cc added chamber
MuonME0DetLayerGeometryBuilder.cc added chamber

GEMSegment.h and ME0Segment.h to use chamberID

fix ME0Geometry.cc layer and chamber functions

clean up GEMSegmentBuilder.cc

MuonTrackLoader.cc added the case for gemSegments

@calabria @kpedro88 @pietverwilligen @archiesharma @dildick

cmsbuild commented:
A new Pull Request was created by  @jshlee (Jason Lee) for CMSSW_8_1_X.

It involves the following packages:

DataFormats/GEMRecHit
Geometry/GEMGeometry
RecoLocalMuon/GEMSegment
RecoMuon/DetLayers
RecoMuon/MuonIdentification
RecoMuon/TrackingTools
SimMuon/GEMDigitizer
Validation/MuonME0Validation


@civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @bellan, @abbiendi, @jhgoh, @calderona, @HuguesBrun, @battibass, @trocino, @bachtis, @rociovilar this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
Please post some performance plots, preferably something that was discussed
already in the POG meeting.

Thanks

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

jshlee commented:
@slava77 - The slides are here.
https://indico.cern.ch/event/586783/contributions/2364321/attachments/1369674/2076730/muonReco_20161110.pdf

Cheers,
Jason.

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

jshlee commented:
changed etaResolution to same as phiResolution
smaller etaResolution is required for granularity studies

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

dildick commented:
@calabria @jshlee why are we changing the etaResolution again? We changed it from 0.03 to 0 only a month ago.

calabria commented:
@dildick because 0 was giving problems in the segment fit... so we decided to put a small value but not exacltly zero

cmsbuild commented:
-1

Tested at: 6ed1c4d26db462d91e12398c7ee1ba27301582c0

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16296/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>In file included from /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-10-1100/src/DataFormats/CLHEP/interface/Migration.h:4:
In file included from /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-10-1100/src/DataFormats/Math/interface/AlgebraicROOTObjects.h:5:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/SVector.h:367:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/BinaryOperators.h:22:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/BinaryOpPolicy.h:22:
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: fatal error: recursive template instantiation exceeded maximum depth of 512
      typedef typename make_indices_impl<I + 1, indices<Indices..., I>,
                       ^
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<507, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 506>, 576>' requested here
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<506, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505>, 576>' requested here
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<505, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504>, 576>' requested here
</pre>

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
-1

Tested at: f12228edb2919f8bd14a33f65f65282daf510380

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16302/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 64 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>In file included from /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-10-1100/src/DataFormats/CLHEP/interface/Migration.h:4:
In file included from /cvmfs/cms-ib.cern.ch/week0/slc6_amd64_gcc530/cms/cmssw-patch/CMSSW_8_1_X_2016-11-10-1100/src/DataFormats/Math/interface/AlgebraicROOTObjects.h:5:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/SVector.h:367:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/BinaryOperators.h:22:
In file included from /cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/BinaryOpPolicy.h:22:
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: fatal error: recursive template instantiation exceeded maximum depth of 512
      typedef typename make_indices_impl<I + 1, indices<Indices..., I>,
                       ^
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<507, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 506>, 576>' requested here
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<506, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505>, 576>' requested here
/cvmfs/cms-ib.cern.ch/2016-46/slc6_amd64_gcc530/lcg/root/6.06.08-giojec2/include/Math/MatrixRepresentationsStatic.h:162:24: note: in instantiation of template class 'ROOT::Math::rowOffsetsUtils::make_indices_impl<505, ROOT::Math::rowOffsetsUtils::indices<0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203, 204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215, 216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227, 228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 250, 251, 252, 253, 254, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275, 276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287, 288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 383, 384, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504>, 576>' requested here
</pre>

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16296/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16302/summary.html

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: b6e80643b41114a11f56301a3753b7336076a6b0


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2746f5d90ed88910fd70bbe740686cce372785a8
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/2746f5d90ed88910fd70bbe740686cce372785a8
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

jshlee commented:
@calabria @pietverwilligen - added the changes to ME0PreRecoGaussianModel so that studies can be done with a resolution of 0.
I added error variables for the case when resolution = 0.

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

dildick commented on SimMuon/GEMDigitizer/src/ME0PreRecoGaussianModel.cc:
```diff
@@ -77,6 +79,8 @@ for (const auto & hit: simHits)
   double tof=CLHEP::RandGaussQ::shoot(engine, hit.timeOfFlight(), sigma_t);
   int pdgid = hit.particleType();
+  if (ex == 0) ex = error_u;//errors cannot be zero
+  if (ey == 0) ey = error_v;
```
Isn't this an error prone operation? Wouldn't it be better to check e.g. ex < 1e-6?

jshlee replied:
should be ok
ex very very small is ok

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16313/summary.html

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 9161fc3f63d4b983e877cc7af6c0137cb5c783da

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16336/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testRecoMETMETProducers had ERRORS

cmsbuild commented:
Comparison job queued.

jshlee commented:
i think the env setting is wrong, ttbarForMetTests.root is made but not found...

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16336/summary.html

slava77 commented:
failure in testRecoMETMETProducers can be ignored indeed.

@smuzaffar is the unit test environment somewhat special? We seem to have this error occasionally for maybe a year or more. And it looks like some lines in RecoMET/METProducers/test/runtests.sh are executed not sequentially or somehow the output file from the first is not seen by the following calls.

smuzaffar commented:
@slava77, unit test env should not be different than any other test. Anyway, I have proposed a change https://github.com/cms-sw/cmssw/pull/16582 here (may be the cmsDriver command was failing)

slava77 commented:
there are quite a few LogErrors with this PR
```
%MSG-e Muon:  StandAloneMuonProducer:refittedStandAloneMuons  16-Nov-2016 ...  Run: 1 Event: 1
ME0s are sorted
```
They occur about 40 times per event in TenMu wflows (e.g. 20011.0)

Please investigate and either fix the cause or if this is benign, reduce the severity to LogInfo

slava77 commented:
There is also a conflict with SimMuon/GEMDigitizer/src/ME0PreRecoGaussianModel.cc

slava77 commented on RecoLocalMuon/GEMSegment/plugins/MuonSegFit.cc:
```diff
@@ -332,7 +326,6 @@ MuonSegFit::SMatrix12by4 MuonSegFit::derivativeMatrix() {
   int row = 0;
   
   for( MuonRecHitContainer::const_iterator it = hits_.begin(); it != hits_.end(); ++it) {
-    if (row > 11) break; // temp due to max size of matrix
```
shouldn't this be put back in with ``` row >= MaxHits2 ``` ?
and also maybe if row is at least MaxHits2 to write a LogError

slava77 commented on RecoMuon/TrackingTools/src/MuonTrackLoader.cc:
```diff
@@ -39,6 +39,7 @@
 
 #include "TrackingTools/Records/interface/TransientRecHitRecord.h"
 
+#include <DataFormats/MuonDetId/interface/GEMDetId.h>
```
please use regular quotes ``` "DataFormats/MuonDetId/interface/GEMDetId.h" ```

slava77 commented:
20011 wf

ME0 chi2 has some more normal values now
![all_sign796-histatvsorig-histat_tenmuextendede2023d1wf20011p0c_log10me0detidme0segmentsownedrangemap_me0segments__reco_obj_collection__data__chi2](https://cloud.githubusercontent.com/assets/4676718/20366037/49973926-abfe-11e6-90dd-384fecdf1167.png)

DOF is chopped at 20
is this expected? (well, I can't obviously point to why it was above 20 before sometimes)
![all_sign796-histatvsorig-histat_tenmuextendede2023d1wf20011p0c_me0detidme0segmentsownedrangemap_me0segments__reco_obj_collection__data__degreesoffreedom](https://cloud.githubusercontent.com/assets/4676718/20366090/7c4c40d2-abfe-11e6-94e9-3551f6a6ae11.png)

![all_sign796-histatvsorig-histat_tenmuextendede2023d1wf20011p0c_me0detidme0segmentsownedrangemap_me0segments__reco_obj_collection__data__localpositionerror_xx](https://cloud.githubusercontent.com/assets/4676718/20366405/d282ea40-abff-11e6-8e66-2c9fd022b516.png)
positionError.xx changes from ~(200 um)**2 to ~(5 um)**2, from something realistic to not realistic. Why? Errors should correspond to real uncertainties.

We do have more muons in ME0 regions (this is good)
![all_sign796-histatvsorig-histat_tenmuextendede2023d1wf20011p0c_recomuons_muons__reco_obj_eta](https://cloud.githubusercontent.com/assets/4676718/20366528/77c78c5e-ac00-11e6-9a6e-0abbdd9bf803.png)

D6 (realistic ME0) has roughly the same behavior

slava77 commented:
After looking at code changes, this appears to be mostly a new feature and not a plain bugfix.
So, this will go to 90X.

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

jshlee commented:
@slava77
The "ME0s are sorted" was a testing msg. should have been removed before.

The ME0 errors are set to really small so that it can be redigi later to study granularity.

slava77 commented:
On 11/17/16 2:19 AM, Jason Lee wrote:
> @slava77 <https://github.com/slava77>
> The "ME0s are sorted" was a testing msg. should have been removed before.

nice

> 
> The ME0 errors are set to really small so that it can be redigi later to
> study granularity.

Please describe how this is going to work
(maybe a sketch of a workflow).
Which inputs will you need for that (can it work from AOD?)

IMHO, the current default high level objects (at least segments, but
maybe even starting with hits) should have the realistic current resolution.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261209188>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbpoqhrtaaqyad-HhED8RXMS6vNLvks5q_Co1gaJpZM4Ktzdl>.
>

davidlange6 commented:
> On Nov 17, 2016, at 2:21 PM, Slava Krutelyov <notifications@github.com> wrote:
> 
> On 11/17/16 2:19 AM, Jason Lee wrote:
> > @slava77 <https://github.com/slava77>
> > The "ME0s are sorted" was a testing msg. should have been removed before.
> 
> nice
> 
> > 
> > The ME0 errors are set to really small so that it can be redigi later to
> > study granularity.
> 
> Please describe how this is going to work
> (maybe a sketch of a workflow).
> Which inputs will you need for that (can it work from AOD?)
> 
> IMHO, the current default high level objects (at least segments, but
> maybe even starting with hits) should have the realistic current resolution.
> 

In my opinion too - but not in the opinion of the upgrade projects and leadership…this already came up in a previous PR

I believe the code you are looking for is

https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc


> 
> > 
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261209188>, or
> > mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AEdcbpoqhrtaaqyad-HhED8RXMS6vNLvks5q_Co1gaJpZM4Ktzdl>.
> > 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

jshlee commented:
Hi @slava77, 

For the ME0 study granularity, it will be run on DIGI files.
The idea is to start from ideal digi hits, so that they can be directly matched to sim hits. 
The ME0 digis will be reDigitiased with different granularities and the errors should be updated correctly here.

@calabria - can you add if I left something out or if there are links to any talks.

slava77 commented:
On 11/17/16 5:40 AM, David Lange wrote:
> 
>> On Nov 17, 2016, at 2:21 PM, Slava Krutelyov <notifications@github.com> wrote:
>> 
>> On 11/17/16 2:19 AM, Jason Lee wrote:
>> > @slava77 <https://github.com/slava77>
>> > The "ME0s are sorted" was a testing msg. should have been removed before.
>> 
>> nice
>> 
>> > 
>> > The ME0 errors are set to really small so that it can be redigi later to
>> > study granularity.
>> 
>> Please describe how this is going to work
>> (maybe a sketch of a workflow).
>> Which inputs will you need for that (can it work from AOD?)
>> 
>> IMHO, the current default high level objects (at least segments, but
>> maybe even starting with hits) should have the realistic current resolution.
>> 
> 
> In my opinion too - but not in the opinion of the upgrade projects and
> leadership…this already came up in a previous PR
> 
> I believe the code you are looking for is
> 
> https://github.com/cms-sw/cmssw/blob/CMSSW_9_0_X/SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc

OK, so the workflow is
run the redigi and then everything else downstream on top of this.

Does this mean we will need to save RECO for the TDR samples?

I'm fine with the precise digis.
I think that the standard downstream collection has to use realistic setup.
That is, the redigi should be a part of the standard workflow in the
RECO step
(which  can be rerun for studies with different parameters at will).



> 
> 
> 
>> 
>> > 
>> > —
>> > You are receiving this because you were mentioned.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261209188>, or
>> > mute the thread
>> > <https://github.com/notifications/unsubscribe-auth/AEdcbpoqhrtaaqyad-HhED8RXMS6vNLvks5q_Co1gaJpZM4Ktzdl>.
>> > 
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or mute the thread.
>> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261248801>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbpQeht1cYzUpe6NlpuDWc69v-wzrks5q_FlhgaJpZM4Ktzdl>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 UCSD: 9500 Gillman Dr, Physics UCSD 0354, La Jolla, CA 92093-0354
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------

slava77 commented:
On 11/17/16 5:43 AM, Jason Lee wrote:
> Hi @slava77 <https://github.com/slava77>,
> 
> For the ME0 study granularity, it will be run on DIGI files.
> The idea is to start from ideal digi hits, so that they can be directly
> matched to sim hits.
> The ME0 digis will be reDigitiased with different granularities and the
> errors should be updated correctly here.

Right now I hear: we want to put wrong unrealistic defaults for all
samples so that we are able to study
*different* realistic versions on a subset of special samples.

Maybe I've heard the message wrong, but it's now in line with the
current implementation.


> 
> @calabria <https://github.com/calabria> - can you add if I left
> something out or if there are links to any talks.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261249463>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbs14LYJsO8mskf6ZbYJrM7Aq-63Hks5q_FoTgaJpZM4Ktzdl>.
>

jshlee committed:
Merged from-CMSSW_8_1_X_2016-11-16-1100 from repository jshlee

jshlee committed:
ME0Segments changed reference detector from etaPartition to chamber

ME0SegmentAlgorithm.cc - changed ME0EtaPartition -> ME0Chamber, bug fix
for failed fits
ME0SegmentAlgorithmBase.h - changed ME0EtaPartition -> ME0Chamber
ME0SegmentBuilder.cc - changed ME0EtaPartition -> ME0Chamber
ME0SegmentMatcher.cc - changed ME0EtaPartition -> ME0Chamber
ME0SegmentsValidation.cc - changed ME0EtaPartition -> ME0Chamber
muonME0DigisPreReco_cfi.py - etaResolution was set to 0, this gave
me0RecHits 0 for y error, which made ME0Segment fitting fail.

MuonSegFit - increased number of hits that can be fitted

MuonGEMDetLayerGeometryBuilder.cc added chamber
MuonME0DetLayerGeometryBuilder.cc added chamber

GEMSegment.h and ME0Segment.h to use chamberID

fix ME0Geometry.cc layer and chamber functions

clean up GEMSegmentBuilder.cc

MuonTrackLoader.cc added the case for gemSegments

jshlee committed:
fix conflict in ME0PreRecoGaussianModel

cmsbuild commented:
Pull request #16536 was updated. @civanch, @Dr15Jones, @cvuosalo, @ianna, @mdhildreth, @dmitrijus, @cmsbuild, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

ianna commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: 2b322b6797fb5aef590859ca6c5d594224289078


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3e64f1b857624755be11f401c71b810822a926fe
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16462/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16462/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16462/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test testJetMETCorrectionsType1MET had ERRORS


* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
23224.0 step2
<pre>runTheMatrix-results/23224.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D5_GenSimHLBeamSpotFull+DigiFull_2023D5+RecoFullGlobal_2023D5+HARVESTFullGlobal_2023D5/step2_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D5_GenSimHLBeamSpotFull+DigiFull_2023D5+RecoFullGlobal_2023D5+HARVESTFullGlobal_2023D5.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/3e64f1b857624755be11f401c71b810822a926fe
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16462/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16536/16462/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

slava77 commented:
@jshlee 
was the last update in this PR just a rebase or did you change the algorithm logic as well?

I'm looking forward for a resolution of the case of unrealistic (precise) ME0 segments and muons

slava77 commented:
I propose that we  continue with 90X copy of this PR and eventually close this 81X version.
(this is a new feature; still incomplete IMO)

for the purpose of continuity, discussion can go on here.

jshlee commented:
@slava77 - I couldn't rebase to fix due to conflict with #16354 in both 81x and 90x...
No changes to algorithm logic, exactly the same as before.
Most of this is a bug fix and clean up.
The only "new" part is the new local point reference frame for ME0 Segments.

The failure seems to be due to hgcalTriggerPrimitiveDigiProducer. 
This PR shouldnt effect step_2.

slava77 commented:
23224.0 is broken in the IB

ianna commented:
-1

as far as I understand, this goes to 90x, not 81x

calabria commented:
@ianna if 81X will be used for the Tracker TDR as planned, it is really important to have these fixes also in 81X, because without them the ME0 segment reco and then the me0muon ID as they are now are completely wrong.

slava77 commented:
I expect that the issue with precise hits/segments (https://github.com/cms-sw/cmssw/pull/16536#issuecomment-261250989) is discussed in the upcoming RECO meeting.

I expect also that it was discussed already in the muon POG meeting and some justification for using precise hits/segments in default workflow is provided. Please provide a link to the slides if the topic was indeed presented.

@calderona, @HuguesBrun

calabria commented:
I don't see the point to have rechits/segments made with the old parameters, anyway this can be done since the ME0 re-digitizer is already merged, but it will not add anything in my opinion apart preserving the old performance (not sure if the differences will be significant). If for you it is ok to add an extra step and save the ideal and "realistic" digis, I am fine with the proposal, we can add the additional step in this PR using the old smearing. Please, consider that the re-digitizer is an improved version of the default pseudodigitizer so there can be some differences: https://indico.cern.ch/event/578192/contributions/2346062/attachments/1357727/2053105/ME0DigiUpdate_20161019.pdf

slava77 commented:
> If for you it is ok to add an extra step and save the ideal and "realistic" digis, I am fine with the proposal, we can add the additional step in this PR using the old smearing

It would be fine for me.

Given the redigitizer availability in the release now and your intent to be able to run tests with re-digitization, I expected that the implementation will be along these lines. (Default products are smeared in a realistic way and ideal digis are kept in the event content for the files/datasets which will be used for the granularity studies).

kpedro88 commented:
@calabria FYI, 81X will not be used for tracker TDR. Phase2 development will continue in 90X until all the TDR components are ready. At that point, we will spin off a release 8_2_0 (based on 90X) to use for TDR sample production.

jshlee commented:
@slava77 - since this release will not be used for the TDR, I will close this and just go with the 90x.
